File: exercice2/java/src/test/java/fr/devoxx/devops/logs/spark/Spark5Test.java
package fr.devoxx.devops.logs.spark;

import fr.devoxx.devops.logs.SparkTest;
import org.apache.spark.util.StatCounter;
import org.junit.Test;
import scala.Tuple4;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.closeTo;
import static org.hamcrest.Matchers.is;

public class Spark5Test extends SparkTest {
    @Test
    public void test_spark() {
        Spark5 test = new Spark5();
        StatCounter stats = test.process(rdd());
        assertThat(stats.count(), is(6102L));
        assertThat(stats.min(), is(40.0));
        assertThat(stats.mean(), closeTo(90.044, 0.001));
        assertThat(stats.max(), is(139.0));
    }

}

File: exercice2/scala/src/test/scala/fr/devoxx/devops/logs/spark/Spark5Test.scala
package fr.devoxx.devops.logs.spark

import fr.devoxx.devops.logs.SharedSparkContext
import org.scalatest.{FunSuite, Matchers}

class Spark5Test extends FunSuite with SharedSparkContext with Matchers {

  test("statistiques sur la taille des requêtes") {
    val result = Spark5(rdd).process
    result.count shouldBe 6102
    result.min shouldBe 40.0
    result.mean shouldBe 90.044 +- 1e-3
    result.max shouldBe 139.0
  }
}
